core/vm/validate.go

```diff
@@ -155,9 +155,6 @@ func validateCode(code []byte, section int, container *Container, jt *JumpTable,
 			if ct := container.ContainerSections[arg]; len(ct.Data) != ct.DataSize {
 				return nil, fmt.Errorf("%w: container %d, have %d, claimed %d, pos %d", ErrEOFCreateWithTruncatedSection, arg, len(ct.Data), ct.DataSize, i)
 			}
-			if _, ok := visitedSubcontainers[arg]; ok {
```

Comment:
Hmm it looks like there is no test in the harness that tests that eofcreating the same subcontainer twice is allowed. I think we should add one (if that is the behavior we want)

Reply:
I took it out because it was causing a test to fail, but I would say it was causing the failure by side-effect not deliberately and by intent. I will make an explicit test to use EOFCREATE and RETURNCONTRACT of the same subcontainer at multiple points in the code.

Reply:
[This test](https://github.com/ethereum/execution-spec-tests/blob/810f2d47a841dea6746c143a6217601287f55b2b/tests/prague/eip7692_eof_v1/eip7620_eof_create/test_eofcreate.py#L458) is re-using the same subcontainer three times, designed to detect address collisions.  So it's tested, just not deliberately as a duplicate reference test.

Reply:
I'm not so sure. When @MariusVanDerWijden took this PR, and remade it against the eof-validation branch, here: https://github.com/holiman/go-ethereum/pull/56/files , one test was affected, but only insofar as going from one type of error into another. 

And those test vectors are taken from the execution-spec reference tests.

Reply:
I added a testvector to execution spec tests that triggers it https://github.com/ethereum/execution-spec-tests/pull/809

Reply:
The collision test didn't have an explicit eof_test, but was a state_test/blockchaiin_test, and that test failed in the EEST fill command.  So it was tested by side effect in the state tests, just not deliberately in the eof_tests, making the error more difficult to find.

Marius's new test makes it explicit in eof_tests in an isolated fashion.